File: leet_1528_shuffle_string.cpp
class Solution {
public:
    string restoreString(string s, vector<int>& indices)
    {
        string  ss = s;
        int i = -1;

        while (++i < s.length())
        {
            // ss[i] = s[indices[i]]; // wrong
            ss[indices[i]] = s[i];
        }
        return ss;
    }
};

File: leet_1528_shuffle_string.py
class Solution:
    def restoreString(self, s: str, indices: List[int]) -> str:
        ss = s
        for i in range(len(s)):
            p = indices[i]
            ss = ss[0 : p] + s[i] + ss[p + 1 : len(s)]
        return ss
